README.md

```diff
@@ -660,6 +660,74 @@ angular.module('AlertListComponentModule', [])
 - Don’t manipulate DOM in your controllers, this will make them harder to test. Use directives instead.
 
 
+### Services and Factories
+
+#### Services should always export an object, not just a single function
+
+_Why_: easier to see at the definition site and the call site exactly what function the service provides
+
+```js
+// Recommended
+angular.module('BuildCSVService', []).factory('BuildCSV', function() {
+  function build() {
+    // ...
+  }
+
+  return {
+    build: build
+  }
+});
+
+// Avoid
+angular.module('BuildCSVService', []).factory('BuildCSV', function() {
+  function build() {
+    // ...
+  }
+
+  return build;
+});
+```
+
+#### Services that take in data and manipulate it should never mutate the original object and return the new object
+
+__Why__: Makes code easier to reason about when it's clearer what is and isn't being mutated.
```

Comment:
Maybe something like `Avoiding mutation in service objects makes it possible to reason about them from their call site without knowing what they do internally.`